File: web/src/services/userProfile.service.ts
import apiClient from '../config/api';
import type { User } from '../types/auth';

export interface UpdateProfilePayload {
  fullName?: string;
  primaryPhone?: string;
  primaryEmail?: string;
  address?: string;
  avatarUrl?: string;
}

export const userProfileService = {
  async updateProfile(payload: UpdateProfilePayload): Promise<User> {
    const response = await apiClient.patch('/users/me', payload);
    return response.data as User;
  },
  async uploadAvatar(file: File): Promise<string> {
    const form = new FormData();
    form.append('avatar', file);
    const response = await apiClient.post('/users/me/avatar', form);
    return response.data.avatarUrl as string;
  },
};

File: mobile/lib/services/user_profile_service.dart
import 'package:dio/dio.dart';
import '../config/api_client.dart';

class UserProfileService {
  final _apiClient = ApiClient();

  Future<Map<String, dynamic>> updateProfile(Map<String, dynamic> payload) async {
    final response = await _apiClient.dio.patch('/users/me', data: payload);
    return Map<String, dynamic>.from(response.data);
  }

  Future<String?> uploadAvatar(String filePath) async {
    final formData = FormData.fromMap({
      'avatar': await MultipartFile.fromFile(filePath, filename: filePath.split('/').last),
    });
    final response = await _apiClient.dio.post('/users/me/avatar', data: formData);
    return response.data['avatarUrl'] as String?;
  }
}

